salt/modules/puppet.py

```diff
@@ -158,7 +163,7 @@ def run(*args, **kwargs):
 
     puppet.kwargs.update(salt.utils.clean_kwargs(**kwargs))
 
-    ret = __salt__['cmd.run_all'](repr(puppet), python_shell=False)
+    ret = __salt__['cmd.run_all'](repr(puppet), python_shell=puppet.useshell)
```

Comment:
Ah, you know what? This is Windows only. I missed that before. I think this is actually fine. Ignore my previous comment.